src/features/presence/PresenceCollaborators.js

```diff
@@ -0,0 +1,118 @@
+import messages from './messages';
+import PresenceAvatar from './PresenceAvatar';
+import { determineInteractionMessage } from './utils/presenceUtils';
+
+import './PresenceCollaborators.scss';
+
+class PresenceCollaborators extends React.Component<Props> {
+    static propTypes = {
+        /* Custom style override for the avatar */
+        avatarStyle: PropTypes.object,
+        collaborators: PropTypes.arrayOf(
+            PropTypes.shape({
+                /** Url to avatar image. If passed in, component will render the avatar image instead of the initials */
+                avatarUrl: PropTypes.string,
+                /** Users id */
+                id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+                isActive: PropTypes.bool,
+                /** Unix timestamp of when the user last interacted with the document */
+                interactedAt: PropTypes.number,
+                /** The type of interaction by the user */
+                interactionType: PropTypes.string,
+                /** User's full name */
+                name: PropTypes.string.isRequired,
+                /** Custom Profile URL */
+                profileUrl: PropTypes.string,
+            }),
+        ).isRequired,
+        /* Intl object */
+        intl: PropTypes.any,
+        /* Custom style override for the list item */
+        itemStyle: PropTypes.object,
+        /* Controls visibility of hover effect for the list item */
+        showHover: PropTypes.bool,
+    };
+
+    state = {
+        selectedCollaborator: null,
+    };
+
+    renderTimestampMessage = (interactedAt, interactionType) => {
+        const { intl } = this.props;
+        const lastActionMessage = determineInteractionMessage(interactionType, interactedAt);
+        const timeAgo = intl.formatRelativeTime
+            ? intl.formatRelativeTime(interactedAt - Date.now())
+            : intl.formatRelative(interactedAt);
+
+        if (lastActionMessage) {
+            return (
+                <FormattedMessage
+                    {...lastActionMessage}
+                    values={{
+                        timeAgo,
+                    }}
+                />
+            );
+        }
+        return null;
+    };
+
+    render() {
+        const { avatarStyle, collaborators, itemStyle, showHover } = this.props;
+        const { selectedCollaborator } = this.state;
+
+        return collaborators.map(collaborator => {
+            const { avatarUrl, id, isActive, interactedAt, interactionType, name, profileUrl } = collaborator;
+
+            const listItemClasses = classnames('bdl-PresenceCollaborators-item', {
+                hover: showHover && selectedCollaborator === collaborator.id,
+            });
+
+            return (
+                <div
+                    key={id}
+                    className={listItemClasses}
+                    onMouseEnter={showHover ? () => this.setState({ selectedCollaborator: id }) : null}
```

Comment:
Can we use the `:hover` pseudo-class, instead? It doesn't seem like the state here is used otherwise.

Reply:
Updated!